src/main/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatchCases.java

```
@@ -107,22 +118,27 @@ static boolean matchDiseases(DataUse purpose, DataUse dataset, Map<String, List<
      *      Any dataset where NMDS is false
      *      Any dataset where NMDS is true AND DS-X match
      */
-    static boolean matchNMDS(DataUse purpose, DataUse dataset, boolean diseaseMatch) {
+    static ImmutablePair<Boolean, List<String>> matchNMDS(DataUse purpose, DataUse dataset, boolean diseaseMatch) {
         // short-circuit if no nmds clause
         if (purpose.getMethodsResearch() == null && dataset.getMethodsResearch() == null) {
-            return true;
+            return ImmutablePair.of(true, Collections.emptyList());
         }
 
+        List<String> failures = new ArrayList<>();
         boolean datasetNMDS = getNullableOrFalse(dataset.getMethodsResearch());
         if (!datasetNMDS) {
-            return true;
+            return ImmutablePair.of(true, Collections.emptyList());
+        } else {
+            failures.add("[NMDS] Dataset is restricted to no method development research");
         }
```

Should we return this failure right away?